items/item-semantics.js

```diff
@@ -74,6 +74,48 @@ class ItemSemantics {
       return null;
     }
   }
+
+  /**
+   * True if the Item is a Location
+   * @type {boolean}
+   */
+  get isLocation () {
+    return Semantics.isLocation(this.rawItem);
+  }
+
+  /**
+   * True if the Item is an Equipment
+   * @type {boolean}
+   */
+  get isEquipment () {
+    return Semantics.isEquipment(this.rawItem);
+  }
+
+  /**
+   * True if the Item is a Point
+   * @type {boolean}
+   */
+  get isPoint () {
+    return Semantics.isPoint(this.rawItem);
+  }
+
+  /**
+   * The Location Item where this Item is situated or `null` if it's not in a Location.
+   * @type {Item|null}
+   */
+  get location () {
+    const rawLoc = Semantics.getLocation(this.rawItem);
+    return (rawLoc) ? items[rawLoc.name] : null;
```

Comment:
```suggestion
    return (rawLoc === null) ? null :  items.getItem(rawLoc.getName());
```

Some explanation:
- Null checks need the `=== null`
- I prefer `items.getItem` inside the library, as it will work if we possibly add some type checking in the future
- `rawLoc.name` is working for now, but using the `getName()` is much cleaner. The property like syntax is called accessors in GraalJS and is only available in Nashorn compatibility mode, which we might disable at some time. Therefore we got and want to get rid of using accessors inside openhab-js.

Reply:
Makes sense. I'll keep that in mind for the future.